src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs

```diff
@@ -542,7 +542,7 @@ private static unsafe int CreateArrayGCDesc(LowLevelList<bool> bitfield, int ran
             return numSeries;
         }
 
-        private static unsafe int CreateGCDesc(LowLevelList<bool> bitfield, int size, bool isValueType, bool isStatic, void* gcdesc)
+        private static unsafe int CreateGCDesc(bool[] bitfield, int size, bool isValueType, bool isStatic, void* gcdesc)
```

Comment:
This method looks unreachable given that `InstanceGCLayout` only returns non-null layout for arrays. Delete it?

Reply:
I believe that the other call is unreachable as well. One of them computes just the size, the other one (that you have deleted here) fills in the memory that was allocated based the size returned from the first call.

Reply:
There is only one place in `GetInstanceGCDescSize` that calls `CreateGCDesc`. My point is that that codepath is unreachable and it can be deleted, in similar way how you deleted it `CreateGCDesc` from `CreateInstanceGCDesc`.

If I am missing something and `CreateGCDesc` is still reachable, it would be a good idea to create a test.

Reply:
Yes, the code within `GetInstanceGCDescSize` also has a dead path. Now the control flow should be the same with `CreateInstanceGCDesc`.